infer_segmentation.py

```diff
@@ -139,7 +141,19 @@ def segment_image(model, image, prediction_threshold, target_size_1d, background
 
 
 def main(gcp_bucket, model_id, background_class_index, stack_id, image_ids, user_specified_prediction_thresholds,
-         labels_output, pad_output, trained_thresholds_id):
+         labels_output, pad_output, trained_thresholds_id, python_random_global_seed, numpy_random_global_seed,
+         tf_random_global_seed):
+
+    # seed global random generators if specified; global random seeds here must be convertible to int or exactly 'None'
+    if python_random_global_seed != 'None':
+        assert isinstance(int(python_random_global_seed), int)
```

Comment:
the obstacle is i can't make argparser type `int` because tho i want default to an int, ialso want to allow `None` as manual input. so to allow numbers and None as input, i need to make type `str` then cast later. technically the seed can of type str, int, float, etc., but i am limiting the type to float or int. the `isinstance` protects against strings other than None that do not convert to `int`, i.e., protects against typos and bounds the input types

Reply:
okay cool got it but I think is a bit confusing way to accomplish that -- so here's my understanding of the different cases:
1. if it's a number (eg `'3'` or `'-1.322'`) this passes
2. if it's other non-number string it errors but not on the assert, it errors because it can't be cast to an int

If I'm understanding, while it is erroring out on non-numbers, it's actually failing because of a `ValueError` not because of the assertion failing (the assertion doesn't even get to finish). 

imo there is a bit of a choice of 3 ways to make this more clear:

1) replaced that assert with just a `int(python_random_global_seed)` since this will produce the same error and behavior but without the extra code of `isinstance` that never runs

2) replaced the assert with: `assert python_random_global_seed.replace('.', '', 1).isdigit()` (this actually checks that it's a number with some `.` removal so it works on floats)

3) not even allow floats since it's deceiving (eg if a user sets a seed of `1.1` and `1.2` they'd likely expect two different sets of random numbers to come of that but since they're both cast to `1` that wouldn't happen). In this case you could then just replace the assert with: `assert python_random_global_seed.isdigit()` (fwiw this one is personally what I'd vote for since it's clearest / minimizes unintuitive behavior)

Reply:
correct, it was failing out before assertion, but i was just thinking the line itself would clarify the issue to people better with the assertion perhaps. happy to remove floats and enforce integers only aka option 3. wasn't sure how restrictive we wanted to be with type, given that seed is flexible, but this works for me.

Reply:
gotcha. yeah I think I see what you're saying but imo would then suggest to explain that in a code comment as opposed to in code